src/core/network-openssl.c

```diff
@@ -648,6 +649,10 @@ static void set_pubkey_info(TLS_REC *tls, X509 *cert, unsigned char *cert_finger
 	bio = BIO_new(BIO_s_mem());
 	ASN1_TIME_print(bio, X509_get_notBefore(cert));
 	length = BIO_read(bio, buffer, sizeof(buffer));
+	if (length < 0) {
+		ret = 0;
+		goto done;
```

Comment:
This leaks bio, same with the other goto.

Reply:
Which way would you prefer to address this? Add `BIO_free(bio);` calls inside the if branches or move `BIO_free(bio);` inside the goto?

Reply:
Either would work, but considering the scope of bio (it refers to two different allocations depending on where you are in the function) it might make more sense to have it inside the ifs.

Reply:
Done, thank you again!